roles/bootstrap-os/tasks/main.yml

```diff
@@ -16,4 +16,15 @@
   register: ostree
 
 - set_fact:
-    is_atomic: "{{ ostree.stat.exists }}"
\ No newline at end of file
+    is_atomic: "{{ ostree.stat.exists }}"
+
+- name: Check configured hostname
+  shell: hostname
+  register: configured_hostname
+  check_mode: no
+
+- name: Assign inventory name to unconfigured hostnames
+  hostname:
+    name: "{{inventory_hostname}}"
+  when: (configured_hostname.stdout == 'localhost')
```

Comment:
This is an incredibly wasteful approach. It should just be:
```
when: ansible_hostname == 'localhost'
```
and get rid of the task "Check configured hostname"